Gemfile

```diff
@@ -103,10 +103,10 @@ group :development, :test do
   gem 'net-http-persistent'
   gem 'rinku'
   gem 'rspec'
-  gem 'selenium-webdriver', '3.141.0'
+  gem 'selenium-webdriver', '~> 4.2.0' # 4.x is required for Ruby 3 support; <4.3 required for Http:Persistent to not be deprecated yet
```

Comment:
FWIW, I had success with selenium-webdriver 4.8 along with:
```
-  require 'selenium/webdriver/remote/http/persistent'
-  class Client < Selenium::WebDriver::Remote::Http::Persistent
+  require 'selenium/webdriver/remote/http/default'
+  class Client < Selenium::WebDriver::Remote::Http::Default
```

`Http::Default` in later Selenium::Webdriver versions uses a persistent http connection (which I believe is why they dropped the Http::Persistent class, it was redundant)